app/our_functions.py

```diff
@@ -0,0 +1,38 @@
+def put_on(personage: dict) -> dict:
+    our_personage = personage.copy()
+    our_personage["protection"] = 0
+    for armour in personage["armour"]:
+        our_personage["protection"] += armour["protection"]
+
+    our_personage["power"] += our_personage["weapon"]["power"]
+
+    if our_personage["potion"] is not None:
+        our_p = our_personage["potion"]
+        if "power" in our_p["effect"]:
+            our_personage["power"] += our_p["effect"]["power"]
+
+        if "protection" in our_p["effect"]:
+            our_personage[
+                "protection"
+            ] += our_p["effect"]["protection"]
+
+        if "hp" in our_p["effect"]:
+            our_personage["hp"] += our_personage["potion"]["effect"]["hp"]
```

Comment:
The handling of potion effects could be more robust. Ensure that all possible effects are correctly applied and consider edge cases where the potion might not have certain effects.